versioned/storage/cache/src/test/java/org/projectnessie/versioned/storage/cache/TestCacheOvershoot.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2025 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.cache;
+
+import static java.util.concurrent.CompletableFuture.delayedExecutor;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.projectnessie.versioned.storage.common.persist.ObjId.randomObjId;
+
+import com.google.common.base.Strings;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.projectnessie.versioned.storage.commontests.objtypes.SimpleTestObj;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public class TestCacheOvershoot {
+  @InjectSoftAssertions protected SoftAssertions soft;
+
+  @RepeatedTest(5)
+  public void testCacheOvershoot() throws Exception {
+    var config =
+        CacheConfig.builder()
+            .capacityMb(4)
+            .cacheCapacityOvershoot(0.1d)
+            // Production uses Runnable::run, but that lets this test sometimes run way too
+            // long, so we introduce some delay to simulate the case that eviction cannot keep up.
+            .executor(t -> delayedExecutor(1, TimeUnit.MILLISECONDS).execute(t))
```

Comment:
Note: this delay let's the test hit the relevant `seen*` assertions pretty quickly.